domain/domain.go

```diff
@@ -382,48 +386,50 @@ func (c *ddlCallback) OnChanged(err error) error {
 	if err != nil {
 		return err
 	}
-	log.Infof("[ddl] on DDL change")
+	log.Infof("[ddl] on DDL change, must reload")
 
-	return c.do.MustReload()
+	return c.do.Reload()
 }
 
 type schemaValidityInfo struct {
 	isValid          bool
-	lastInvalidTS    uint64 // It's used for recording the last txn TS of schema invalid.
+	firstValidTS     uint64 // It's used for recording the first txn TS of schema vaild.
 	mux              sync.RWMutex
-	lastFailedTS     uint64 // It's used for recording the last txn TS of loading schema failed.
+	lastReloadTime   int64  // It's used for recording the time of last reload schema.
+	lastSuccTS       uint64 // It's used for recording the last txn TS of loading schema succeed.
 	MockReloadFailed bool   // It mocks reload failed.
 }
 
-func (s *schemaValidityInfo) setLastFailedTS(ts uint64) {
-	atomic.StoreUint64(&s.lastFailedTS, ts)
+func (s *schemaValidityInfo) updateTimeInfo(lastReloadTime int64, lastSuccTS uint64) {
+	s.mux.Lock()
+	defer s.mux.Unlock()
+
+	s.lastReloadTime = lastReloadTime
+	s.lastSuccTS = lastSuccTS
 }
 
-func (s *schemaValidityInfo) getLastFailedTS() uint64 {
-	return atomic.LoadUint64(&s.lastFailedTS)
+func (s *schemaValidityInfo) getTimeInfo() (int64, uint64) {
+	s.mux.RLock()
+	defer s.mux.RUnlock()
+
+	return s.lastReloadTime, s.lastSuccTS
 }
 
 // SetValidity sets the schema validity value.
 // It's public in order to do the test.
-func (s *schemaValidityInfo) SetValidity(v bool) {
+func (s *schemaValidityInfo) SetValidity(v bool, lastSuccTS uint64) {
```

Comment:
Can we remove the `lastSuccTS` parameter, and set `s.firstValidTS = s.lastSuccTS` directly?